presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java

```diff
@@ -523,4 +523,10 @@ protected static void testIncompactBlock(Block block)
         assertNotCompact(block);
         testCopyRegionCompactness(block);
     }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T[] newArrayOfType(T[] array, int length)
```

Comment:
I would directly call `java.util.Arrays#copyOf(T[], int)`. Optionally followed by `Arrays.fill(.., null)` (which is not strictly necessary, cause you set all the elements anyway)

Reply:
Good idea, I'll update to that